common/Geometry2d/Rect.cpp

```diff
@@ -18,6 +18,111 @@ bool Rect::intersects(const Rect& other) const {
     }
 }
 
+int Rect::CohenSutherlandCode(const Point& other) const{
```

Comment:
I'm on the fence about the naming for this function. At first glance it felt like just saying it's the Cohen Sutherland C++ code, but after looking at the algorithm, I see it's referring to the out code. Making it `CohenSutherlandOutCode`, `ComputeOutCode` etc may make it clearer.

Reply:
That seems doable

Reply:
refactored to CohenSutherlandOutCode